dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java

```diff
@@ -0,0 +1,313 @@
+package com.dotcms.browser;
+
+import static org.junit.Assert.assertTrue;
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.contenttype.exception.NotFoundInDbException;
+import com.dotcms.datagen.FileAssetDataGen;
+import com.dotcms.datagen.FolderDataGen;
+import com.dotcms.datagen.HTMLPageDataGen;
+import com.dotcms.datagen.LanguageDataGen;
+import com.dotcms.datagen.LinkDataGen;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.image.focalpoint.FocalPointAPITest;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.IndexPolicy;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.UUIDGenerator;
+import com.google.common.collect.ImmutableSet;
+import com.liferay.portal.model.User;
+import io.vavr.Tuple;
+import io.vavr.Tuple3;
+
+/**
+ * Created by Oscar Arrieta on 6/8/17.
+ */
+
+public class BrowserAPITest extends IntegrationTestBase {
+
+    final BrowserAPI browserAPI = APILocator.getBrowserAPI();
+    final FolderAPI folderAPI = APILocator.getFolderAPI();
+    final UserAPI userAPI = APILocator.getUserAPI();
+    final HostAPI hostAPI = APILocator.getHostAPI();
+
+    static Host testHost;
+    static Folder testFolder, testSubFolder;
+    static HTMLPageAsset testPage;
+    static Language testLanguage;
+    static Contentlet testDotAsset;
+    static FileAsset testFileAsset, testFileAsset2, testFileAsset3Archived, testFileAsset2MultiLingual;
+
+    static Link testlink;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)